tools/cmake/utility.cmake

```
@@ -43,26 +43,42 @@ function(set_macro_definitions)
                         ${application_target} PRIVATE
                             ${required_macro_definition}="${${required_macro_definition}}"
                     )
-                    message("Defining ${required_macro_definition} to ${${required_macro_definition}} in ${application_target}")
                     # This variable adds definitions to the file being run against `check_symbol_exists`.
                     list(APPEND CMAKE_REQUIRED_DEFINITIONS -D${required_macro_definition})
-                    list(APPEND DEFINED_MACROS "${required_macro_definition}")
+                    list(APPEND DEFINED_MACROS_FROM_CMAKE "${required_macro_definition}")
                     continue()
                 endif()
                 # Check if MACRO_DEFINITIONS_FILES_TO_CHECK has the required macros defined already.
                 get_target_property(APPLICATION_INCLUDES ${application_target} INCLUDE_DIRECTORIES)
                 set(CMAKE_REQUIRED_INCLUDES ${APPLICATION_INCLUDES})
+                set(CMAKE_REQUIRED_QUIET TRUE)
                 unset(HAVE_${required_macro_definition} CACHE)
                 check_symbol_exists(${required_macro_definition} ${MACRO_DEFINITIONS_FILES_TO_CHECK} HAVE_${required_macro_definition})
                 # Append to the right list depending on whether definition was found.
                 if(HAVE_${required_macro_definition})
-                message("Using value provided by ${MACRO_DEFINITIONS_FILES_TO_CHECK} for ${required_macro_definition} in ${application_target}")
-                    list(APPEND DEFINED_MACROS "${required_macro_definition}")
+                    list(APPEND DEFINED_MACROS_FROM_FILE "${required_macro_definition}")
                 else()
                     list(APPEND MISSING_REQUIRED_MACROS "${required_macro_definition}")
                 endif()
             endforeach()
         endif()
+        # Log the file of the defined macros
+        list(LENGTH DEFINED_MACROS_FROM_FILE DEFINED_MACROS_FROM_FILE_LENGTH)
+        if(DEFINED_MACROS_FROM_FILE_LENGTH GREATER 0)
+            message("Using values in ${MACRO_DEFINITIONS_FILES_TO_CHECK} to define the following macros for ${application_target}:")
+            foreach(defined_macro_from_file ${DEFINED_MACROS_FROM_FILE})
+                message("${defined_macro_from_file}")
+            endforeach()
+        endif()
+        # Log the value of the macros passed through CMake
+        list(LENGTH DEFINED_MACROS_FROM_CMAKE DEFINED_MACROS_FROM_CMAKE_LENGTH)
+        if(DEFINED_MACROS_FROM_CMAKE_LENGTH GREATER 0)
+            message("Passing the following CMake arguments as macros for ${application_target}:")
```

(Minor) Suggestion to keep the sentence similar to the previous one 🙂
```suggestion
            message("Using the passed CMake arguments to define the following macros for ${application_target}:")
```